renderer.go

```diff
@@ -59,7 +67,16 @@ func (r *Renderer) SetOutput(o *termenv.Output) {
 
 // ColorProfile returns the detected termenv color profile.
 func (r *Renderer) ColorProfile() termenv.Profile {
-	return r.output.Profile
+	r.mtx.RLock()
+	defer r.mtx.RUnlock()
+
+	if !r.explicitColorProfile {
+		r.getColorProfile.Do(func() {
+			r.colorProfile = r.output.EnvColorProfile()
```

Comment:
should it wlock here?

Reply:
It looks like we do, but I think we don't need to since the function would only run once. Maybe @muesli can weigh on this

Reply:
Agree, the `sync.Once` already provides its own locking mechanism, so the read-lock should be sufficient. Could maybe add a comment in that regard to prevent further confusion.